src/main/java/org/opensrp/connector/dhis2/DHIS2ImportOrganizationUnits.java

```diff
@@ -0,0 +1,343 @@
+package org.opensrp.connector.dhis2;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import org.apache.commons.lang3.StringUtils;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.opensrp.domain.AppStateToken;
+import org.opensrp.repository.AppStateTokensRepository;
+import org.opensrp.service.LocationTagService;
+import org.opensrp.service.PhysicalLocationService;
+import org.slf4j.LoggerFactory;
+import org.smartregister.domain.Geometry;
+import org.smartregister.domain.LocationProperty;
+import org.smartregister.domain.LocationTag;
+import org.smartregister.domain.PhysicalLocation;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.*;
+
+import static org.opensrp.connector.dhis2.DHIS2Constants.*;
+import static org.opensrp.connector.dhis2.DHISUtils.parseDhisDate;
+import static org.opensrp.connector.dhis2.Dhis2EndPoints.*;
+
+@Component
+public class DHIS2ImportOrganizationUnits extends DHIS2Service {
+
+	private final AppStateTokensRepository allAppStateTokens;
+
+	@Autowired
+	private PhysicalLocationService physicalLocationService;
+
+	@Autowired
+	private LocationTagService locationTagService;
+
+	private Gson gson = new Gson();
+
+	private Integer rowsProcessed = 0;
+
+	private Boolean isJobRunning = false;
+
+	private Boolean isJobFailed = false;
+
+	private static org.slf4j.Logger logger = LoggerFactory.getLogger(DHIS2ImportOrganizationUnits.class.toString());
+
+	public DHIS2ImportOrganizationUnits(AppStateTokensRepository allAppStateTokens) {
+		this.allAppStateTokens = allAppStateTokens;
+	}
+
+	@Async
+	public void importOrganizationUnits(String startPage) {
+
+		try {
+			JSONObject root = getOrganisationalUnitList(Integer.parseInt(startPage));
+			JSONObject pager = (JSONObject) root.get("pager");
+
+			while (true) {
+				isJobRunning = Boolean.TRUE;
+				JSONArray orgUnits = (JSONArray) root.get(ORG_UNIT_KEY);
+				for (int i = 0; i < orgUnits.length(); i++) {
+				createOrUpdatePhysicalLocation(orgUnits.getJSONObject(i), pager);
+				}
+
+				if (pager.has(ORG_UNIT_NEXT_PAGE_KEY)) {
+					Integer currentPageNumber = (Integer) pager.get(ORG_UNIT_PAGE_KEY);
+					System.out.println("Going to page number  : " + currentPageNumber + 1);
+					root = getOrganisationalUnitList(currentPageNumber + 1);
```

Comment:
This **if** check is inside the while loop, and if there is a next page key present, the root JSON object is updated with the updated page's data and the Current Page Number object also updates. This will work recursively until there is no key of the next page found, the loop will break when there is no key found of the next page.